### PR TITLE
eslint, with very conservative initial rules

### PR DESCRIPTION
**Pull request by Pomax**
removed the gulp jshint+jscs tasks, replaced them with an eslint npm task instead. Initial rules are only the semicolon and strict rules from mofo-style, so we can start iteratively getting the code in line with mofo-style without completely breaking the build.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "ecmaFeatures": {
+      "jsx": true
+    },
+  },
+  "rules" : {
+    "semi": [2, "always"],
+    "strict": [2, "global"]
+  },
+  "extends": [
+    // "./node_modules/mofo-style/.eslintrc.yaml"
```
Comment (Pomax) [inline .eslintrc]: one day soon, this will get uncommented, and it will be a good day.

File: package.json [hunk 2]
```diff
@@ -17,9 +17,11 @@
   },
   "scripts": {
     "app": "npm run l10n && gulp app",
-    "build": "gulp",
-    "l10n": "pontoon-to-json",
+    "build": "npm run lint && gulp",
```
Comment (gvn) [inline package.json]: Having a test as part of the build task seems a bit weird to me. Won't Travis run the linting task twice this way?

File: package.json [hunk 3]
```diff
@@ -17,9 +17,11 @@
...
     "cleanup": "rm -rf dist && npm run l10n",
+    "l10n": "pontoon-to-json",
+    "lint": "eslint --ext .js --ext .jsx .",
```
Comment (gvn) [inline package.json]: Why not make this `test:eslint`?

Comment (gvn) [inline package.json]: Then we can eventually add `npm-run-all` and just run all the test:** tasks for `test`.

File: .eslintrc (hunk 1 above)
Comment (gvn) [inline .eslintrc]: I guess this file is either JSON that allows comments or unvalidated JSON?

**Comment by gvn** [comment]
Some thoughts on task naming, but LGTM!

File: package.json (hunk 3 above)
Comment (Pomax) [inline package.json]: good idea.

File: package.json (hunk 2 above)
Comment (Pomax) [inline package.json]: yeah I've taken this task out in one of the follow ups. There... isn't actually anything that uses it! \o/

File: .eslintrc (hunk 1 above)
Comment (Pomax) [inline .eslintrc]: it's super nice eslint "json but then with comments" (:heart:)

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    },
```
Comment (alanmoo) [inline .eslintrc]: This file doesn't pass eslint because of the comma here 😆 😜

File: package.json [hunk 5]
```diff
@@ -28,7 +30,7 @@
     "postinstall": "npm run l10n && gulp postinstall",
     "smoketest": "gulp smoketest",
     "start": "npm run app",
-    "test": "gulp smoketest && npm run test:mocha && npm run test:browser && gulp lint-test",
+    "test": "npm run lint && gulp smoketest && npm run test:mocha && npm run test:browser",
```
Comment (alanmoo) [inline package.json]: Finally. 👍

**Comment by alanmoo** [comment]
LGTM too

File: .eslintrc (hunk 4 above)
Comment (Pomax) [inline .eslintrc]: it's okay, it's not technically JSON... I think =D

**Comment by Pomax** [comment]
closing as reviewed and superceded.

**Pull request closed without merging**